tutorials/unifiers/src/test/java/com/example/myapexapp/ApplicationTest.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Put your copyright and license info here.
+ */
+package com.example.myapexapp;
+
+import java.io.IOException;
+
+import javax.validation.ConstraintViolationException;
+
+import org.junit.Assert;
+
+import org.apache.hadoop.conf.Configuration;
+import org.junit.Test;
+
+import com.datatorrent.api.LocalMode;
+import com.example.myapexapp.Application;
+
+/**
+ * Test the DAG declaration in local mode.
+ */
+public class ApplicationTest {
+
+  // helper routine
+  private void go(final boolean useUnifier) throws Exception {
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml"));
+      if (useUnifier) {
+        conf.setBoolean("dt.application.MyFirstApplication.operator.range.prop.useUnifier",
+                        true);
+      }
+      lma.prepareDAG(new Application(), conf);
+      LocalMode.Controller lc = lma.getController();
+      lc.run(10000); // runs for 10 seconds and quits
```

Comment:
Thought about that but couldn't see an easy way of doing it (didn't want to
complicate the application by adding file writers just for the sake of unit
tests).

Any suggestions ?

On Wed, Jun 8, 2016 at 7:57 PM, Thomas Weise <notifications@github.com>
wrote:

> In
> tutorials/unifiers/src/test/java/com/example/myapexapp/ApplicationTest.java
> <https://github.com/DataTorrent/examples/pull/11#discussion_r66376084>:
>
> > + */
> > +public class ApplicationTest {
> > +
> > +  // helper routine
> > +  private void go(final boolean useUnifier) throws Exception {
> > +    try {
> > +      LocalMode lma = LocalMode.newInstance();
> > +      Configuration conf = new Configuration(false);
> > +      conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties.xml"));
> > +      if (useUnifier) {
> > +        conf.setBoolean("dt.application.MyFirstApplication.operator.range.prop.useUnifier",
> > +                        true);
> > +      }
> > +      lma.prepareDAG(new Application(), conf);
> > +      LocalMode.Controller lc = lma.getController();
> > +      lc.run(10000); // runs for 10 seconds and quits
>
> In the future we want to validate the result instead of doing this!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DataTorrent/examples/pull/11/files/43cefb65d55ddec7c2cc6d202129efdf8305000b#r66376084>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACYeiZx-Un4bNoORg2Bi1dcnvvN_-fn8ks5qJ4EHgaJpZM4IxlSX>
> .
>